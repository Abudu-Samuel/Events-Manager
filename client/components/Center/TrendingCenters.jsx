import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const TrendingCenter = ({
  centers,
  getCenterId,
}) => (
  <div className="container">
    <section id="popular centers">
      <h2 className="mb-3 slot font-weight-bold text-center">Latest Centers</h2>
      <div className="row mb-4">
        {
          centers.center.map(center => (<div className="col-md-4 mb-4"
            key={center.id}>
            <div className="card hoverable text-center">
              <img className="img-fluid hoverable max" src={center.image}
                alt="Card image cap" />
              <div className="card-body">
                <h4 className="card-title">{center.name}</h4>
                <p className="card-text">
                  {
                    center.description.split('').length > 40 ?
                      `${center.description.slice(0, 40)}...` :
                      center.description
                  }
                </p>
                <button
                  id="viewCenter"
                  className="btn btn-mycolor btn-sm"
                  data-centerid={center.id}
                  onClick={getCenterId}>
                  <Link to={`/centers/${center.id}`}>Details</Link>
                </button>
              </div>
            </div>
          </div>))
        }
      </div>
    </section>
  </div>
);

TrendingCenter.propTypes = {
  centers: PropTypes.object,
  getCenterId: PropTypes.func
};

export default TrendingCenter;
