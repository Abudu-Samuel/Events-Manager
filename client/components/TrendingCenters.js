import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import Navbar from './Navbar';

import * as userActions from '../actions/actionCreator';

class TrendingCenter extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            user: null,
            centers: [],
            fetchingCenters: false
        };
    }

    componentWillMount() {
        this.props.getAllCenters();
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.getCenters[0]) {
            this.setState({ centers: nextProps.getCenters, fetchingCenters: false });
        } else {
            this.setState({ fetchingCenters: true });
        }
    }
    /**
     *
     *
     * @returns
     *
     * @memberOf TrendingCenter
     */
    render() {
        return (
            <div className="container space">
                <Navbar />
                <section id="popular centers">
                    <h2 className="mb-3 font-weight-bold grey-text">Trending Centers</h2>
                    <div className="row mb-4">
                        {
                            this.state.centers.map((center, key) => (<div className="col-md-4 mb-4" key={center.id}>
                                    <div className="card text-center">
                                        <img className="img-fluid hoverable" src={center.image} alt="Card image cap" />
                                        <div className="card-body">
                                            <h4 className="card-title">{center.name}</h4>
                                            <p className="card-text">{center.description}</p>
                                            <Link to="/centerdetails" className="btn btn-mycolor btn-sm">Details</Link>
                                        </div>
                                    </div>
                                </div>))
                        }
                    </div>
                </section>
                <hr className="my-5" />
            </div>
        );
    }
}


function mapStateToProps(state, ownProps) {
    return {
        getCenters: state.centers.foundCenters
    };
}

function mapDispatchToProps(dispatch) {
    return {
        getAllCenters: (centerData) => dispatch(userActions.getAllCenters(centerData))
    };
}

const TrendingCenters = connect(mapStateToProps, mapDispatchToProps)(TrendingCenter);

export default TrendingCenters;
