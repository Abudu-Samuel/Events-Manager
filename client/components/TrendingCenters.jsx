import React from 'react';
import { Link } from 'react-router-dom';
import Navbar from './Navbar';


const TrendingCenter = ({
    centers
}) => (
            <div className="container space">
                <Navbar />
                <section id="popular centers">
                    <h2 className="mb-3 font-weight-bold grey-text">Trending Centers</h2>
                    <div className="row mb-4">
                        {
                            centers.map((center, key) => (<div className="col-md-4 mb-4" key={center.id}>
                                    <div className="card text-center">
                                        <img className="img-fluid hoverable" src={center.image} alt="Card image cap" />
                                        <div className="card-body">
                                            <h4 className="card-title">{center.name}</h4>
                                            <p className="card-text">{center.description}</p>
                                            <Link to="/centerdetails" className="btn btn-mycolor btn-sm">Details</Link>
                                        </div>
                                    </div>
                                </div>))
                        }
                    </div>
                </section>
                <hr className="my-5" />
            </div>
        );

export default TrendingCenter;
