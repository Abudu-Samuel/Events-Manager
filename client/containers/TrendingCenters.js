import React from 'react';

/**
 * creates Navbar component
 * @returns {funct} Trending Center
 */
const TrendingCenter = () => (
    <div className="container">
        <section id="popular centers">
            <h2 className="mb-5 font-weight-bold text-center">Trending Centers</h2>
            <div className="row mb-4">
                <div className="col-md-4 mb-4">
                    <div className="card text-center">
                        <img className="img-fluid hoverable" src="https://static.pexels.com/photos/207731/pexels-photo-207731.jpeg" alt="Card image cap" />
                        <div className="card-body">
                            <h4 className="card-title">Hall Of Fame</h4>
                            <p className="card-text">Some quick example text to build on the card title and make up the bulk of the card's content.</p>
                            <a href="centerdetails.html" className="btn btn-mycolor btn-sm">info</a>
                        </div>
                    </div>
                </div>
                <div className="col-md-4 mb-4">
                    <div className="card text-center">
                        <img className="img-fluid hoverable" src="https://static.pexels.com/photos/69815/florida-state-university-westcott-building-auditorium-interior-69815.jpeg" alt="Card image cap" />
                        <div className="card-body">
                            <h4 className="card-title">Hall Of Fame</h4>
                            <p className="card-text">Some quick example text to build on the card title and make up the bulk of the card's content.</p>
                            <a href="centerdetails.html" className="btn btn-mycolor btn-sm">info</a>
                        </div>
                    </div>
                </div>
                <div className="col-md-4 mb-4">
                    <div className="card text-center">
                        <img className="img-fluid hoverable" src="https://static.pexels.com/photos/260689/pexels-photo-260689.jpeg" alt="Card image cap" />
                        <div className="card-body">
                            <h4 className="card-title">Hall Of Fame</h4>
                            <p className="card-text">Some quick example text to build on the card title and make up the bulk of the card's content.</p>
                            <a href="centerdetails.html" className="btn btn-mycolor btn-sm">info</a>
                        </div>
                    </div>
                </div>
            </div>
        </section>
        <hr className="my-5" />
    </div>
);

export default TrendingCenter;
