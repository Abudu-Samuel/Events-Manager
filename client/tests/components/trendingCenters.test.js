import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import TrendingCenters from '../../components/Center/TrendingCenters';
import centerMockedData from '../__mocks__/centerMockedData';


const props = {
  centers: {
    center: [centerMockedData.centerData]
  }
};

describe('<TrendingCenters />', () => {
  it('should render Footer correctly', () => {
    const wrapper = shallow(<TrendingCenters {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
