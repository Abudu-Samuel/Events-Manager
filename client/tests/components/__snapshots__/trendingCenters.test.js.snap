// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrendingCenters /> should render Footer correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrendingCenter
    centers={
      Object {
        "center": Array [
          Object {
            "capacity": 123,
            "description": "Awesome center",
            "image": "testing.img",
            "isAvailable": true,
            "location": "yaba",
            "name": "test center",
            "price": 123456,
            "state": "lagos",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        id="popular centers"
      >
        <h2
          className="mb-3 slot font-weight-bold text-center"
        >
          Latest Centers
        </h2>
        <div
          className="row mb-4"
        >
          <div
            className="col-md-4 mb-4"
          >
            <div
              className="card hoverable text-center"
            >
              <img
                className="img-fluid hoverable max"
                src="testing.img"
              />
              <div
                className="card-body"
              >
                <h4
                  className="card-title"
                >
                  test center
                </h4>
                <p
                  className="card-text"
                >
                  Awesome center
                </p>
                <button
                  className="btn btn-mycolor btn-sm"
                  id="viewCenter"
                >
                  <Link
                    replace={false}
                    to="/centers/undefined"
                  >
                    Details
                  </Link>
                </button>
              </div>
            </div>
          </div>
        </div>
      </section>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="mb-3 slot font-weight-bold text-center"
          >
            Latest Centers
          </h2>,
          <div
            className="row mb-4"
          >
            <div
              className="col-md-4 mb-4"
            >
              <div
                className="card hoverable text-center"
              >
                <img
                  className="img-fluid hoverable max"
                  src="testing.img"
                />
                <div
                  className="card-body"
                >
                  <h4
                    className="card-title"
                  >
                    test center
                  </h4>
                  <p
                    className="card-text"
                  >
                    Awesome center
                  </p>
                  <button
                    className="btn btn-mycolor btn-sm"
                    id="viewCenter"
                  >
                    <Link
                      replace={false}
                      to="/centers/undefined"
                    >
                      Details
                    </Link>
                  </button>
                </div>
              </div>
            </div>
          </div>,
        ],
        "id": "popular centers",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Latest Centers",
            "className": "mb-3 slot font-weight-bold text-center",
          },
          "ref": null,
          "rendered": "Latest Centers",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-md-4 mb-4"
              >
                <div
                  className="card hoverable text-center"
                >
                  <img
                    className="img-fluid hoverable max"
                    src="testing.img"
                  />
                  <div
                    className="card-body"
                  >
                    <h4
                      className="card-title"
                    >
                      test center
                    </h4>
                    <p
                      className="card-text"
                    >
                      Awesome center
                    </p>
                    <button
                      className="btn btn-mycolor btn-sm"
                      id="viewCenter"
                    >
                      <Link
                        replace={false}
                        to="/centers/undefined"
                      >
                        Details
                      </Link>
                    </button>
                  </div>
                </div>
              </div>,
            ],
            "className": "row mb-4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="card hoverable text-center"
                >
                  <img
                    className="img-fluid hoverable max"
                    src="testing.img"
                  />
                  <div
                    className="card-body"
                  >
                    <h4
                      className="card-title"
                    >
                      test center
                    </h4>
                    <p
                      className="card-text"
                    >
                      Awesome center
                    </p>
                    <button
                      className="btn btn-mycolor btn-sm"
                      id="viewCenter"
                    >
                      <Link
                        replace={false}
                        to="/centers/undefined"
                      >
                        Details
                      </Link>
                    </button>
                  </div>
                </div>,
                "className": "col-md-4 mb-4",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <img
                      className="img-fluid hoverable max"
                      src="testing.img"
                    />,
                    <div
                      className="card-body"
                    >
                      <h4
                        className="card-title"
                      >
                        test center
                      </h4>
                      <p
                        className="card-text"
                      >
                        Awesome center
                      </p>
                      <button
                        className="btn btn-mycolor btn-sm"
                        id="viewCenter"
                      >
                        <Link
                          replace={false}
                          to="/centers/undefined"
                        >
                          Details
                        </Link>
                      </button>
                    </div>,
                  ],
                  "className": "card hoverable text-center",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "img-fluid hoverable max",
                      "src": "testing.img",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <h4
                          className="card-title"
                        >
                          test center
                        </h4>,
                        <p
                          className="card-text"
                        >
                          Awesome center
                        </p>,
                        <button
                          className="btn btn-mycolor btn-sm"
                          id="viewCenter"
                        >
                          <Link
                            replace={false}
                            to="/centers/undefined"
                          >
                            Details
                          </Link>
                        </button>,
                      ],
                      "className": "card-body",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "test center",
                          "className": "card-title",
                        },
                        "ref": null,
                        "rendered": "test center",
                        "type": "h4",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Awesome center",
                          "className": "card-text",
                        },
                        "ref": null,
                        "rendered": "Awesome center",
                        "type": "p",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <Link
                            replace={false}
                            to="/centers/undefined"
                          >
                            Details
                          </Link>,
                          "className": "btn btn-mycolor btn-sm",
                          "data-centerid": undefined,
                          "id": "viewCenter",
                          "onClick": undefined,
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "children": "Details",
                            "replace": false,
                            "to": "/centers/undefined",
                          },
                          "ref": null,
                          "rendered": "Details",
                          "type": [Function],
                        },
                        "type": "button",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          id="popular centers"
        >
          <h2
            className="mb-3 slot font-weight-bold text-center"
          >
            Latest Centers
          </h2>
          <div
            className="row mb-4"
          >
            <div
              className="col-md-4 mb-4"
            >
              <div
                className="card hoverable text-center"
              >
                <img
                  className="img-fluid hoverable max"
                  src="testing.img"
                />
                <div
                  className="card-body"
                >
                  <h4
                    className="card-title"
                  >
                    test center
                  </h4>
                  <p
                    className="card-text"
                  >
                    Awesome center
                  </p>
                  <button
                    className="btn btn-mycolor btn-sm"
                    id="viewCenter"
                  >
                    <Link
                      replace={false}
                      to="/centers/undefined"
                    >
                      Details
                    </Link>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </section>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="mb-3 slot font-weight-bold text-center"
            >
              Latest Centers
            </h2>,
            <div
              className="row mb-4"
            >
              <div
                className="col-md-4 mb-4"
              >
                <div
                  className="card hoverable text-center"
                >
                  <img
                    className="img-fluid hoverable max"
                    src="testing.img"
                  />
                  <div
                    className="card-body"
                  >
                    <h4
                      className="card-title"
                    >
                      test center
                    </h4>
                    <p
                      className="card-text"
                    >
                      Awesome center
                    </p>
                    <button
                      className="btn btn-mycolor btn-sm"
                      id="viewCenter"
                    >
                      <Link
                        replace={false}
                        to="/centers/undefined"
                      >
                        Details
                      </Link>
                    </button>
                  </div>
                </div>
              </div>
            </div>,
          ],
          "id": "popular centers",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Latest Centers",
              "className": "mb-3 slot font-weight-bold text-center",
            },
            "ref": null,
            "rendered": "Latest Centers",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col-md-4 mb-4"
                >
                  <div
                    className="card hoverable text-center"
                  >
                    <img
                      className="img-fluid hoverable max"
                      src="testing.img"
                    />
                    <div
                      className="card-body"
                    >
                      <h4
                        className="card-title"
                      >
                        test center
                      </h4>
                      <p
                        className="card-text"
                      >
                        Awesome center
                      </p>
                      <button
                        className="btn btn-mycolor btn-sm"
                        id="viewCenter"
                      >
                        <Link
                          replace={false}
                          to="/centers/undefined"
                        >
                          Details
                        </Link>
                      </button>
                    </div>
                  </div>
                </div>,
              ],
              "className": "row mb-4",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="card hoverable text-center"
                  >
                    <img
                      className="img-fluid hoverable max"
                      src="testing.img"
                    />
                    <div
                      className="card-body"
                    >
                      <h4
                        className="card-title"
                      >
                        test center
                      </h4>
                      <p
                        className="card-text"
                      >
                        Awesome center
                      </p>
                      <button
                        className="btn btn-mycolor btn-sm"
                        id="viewCenter"
                      >
                        <Link
                          replace={false}
                          to="/centers/undefined"
                        >
                          Details
                        </Link>
                      </button>
                    </div>
                  </div>,
                  "className": "col-md-4 mb-4",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <img
                        className="img-fluid hoverable max"
                        src="testing.img"
                      />,
                      <div
                        className="card-body"
                      >
                        <h4
                          className="card-title"
                        >
                          test center
                        </h4>
                        <p
                          className="card-text"
                        >
                          Awesome center
                        </p>
                        <button
                          className="btn btn-mycolor btn-sm"
                          id="viewCenter"
                        >
                          <Link
                            replace={false}
                            to="/centers/undefined"
                          >
                            Details
                          </Link>
                        </button>
                      </div>,
                    ],
                    "className": "card hoverable text-center",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "img-fluid hoverable max",
                        "src": "testing.img",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <h4
                            className="card-title"
                          >
                            test center
                          </h4>,
                          <p
                            className="card-text"
                          >
                            Awesome center
                          </p>,
                          <button
                            className="btn btn-mycolor btn-sm"
                            id="viewCenter"
                          >
                            <Link
                              replace={false}
                              to="/centers/undefined"
                            >
                              Details
                            </Link>
                          </button>,
                        ],
                        "className": "card-body",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "test center",
                            "className": "card-title",
                          },
                          "ref": null,
                          "rendered": "test center",
                          "type": "h4",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Awesome center",
                            "className": "card-text",
                          },
                          "ref": null,
                          "rendered": "Awesome center",
                          "type": "p",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": <Link
                              replace={false}
                              to="/centers/undefined"
                            >
                              Details
                            </Link>,
                            "className": "btn btn-mycolor btn-sm",
                            "data-centerid": undefined,
                            "id": "viewCenter",
                            "onClick": undefined,
                          },
                          "ref": null,
                          "rendered": Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "class",
                            "props": Object {
                              "children": "Details",
                              "replace": false,
                              "to": "/centers/undefined",
                            },
                            "ref": null,
                            "rendered": "Details",
                            "type": [Function],
                          },
                          "type": "button",
                        },
                      ],
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
